### DISCUSSION TITLE
Question : Do you have example how to make the chart animated ?

### DISCUSSION THREAD

**Discussion by x4080** [mhkeller/layercake#205] [Q&A] [answered]
Do you have example how to make the chart animated ? for example I have initial data :
```
data = [
{ x: "john", y: 100 },
{ x: "mary", y: 200 },
{ x: "keanu", y: 320 },
];
```

then data updated to

```
data = [
{ x: "john", y: 1000 },
{ x: "mary", y: 2000 },
{ x: "keanu", y: 3200 },
];
```

the chart would animate from initial data to updated data

Thanks

**Comment by mhkeller** [comment, marked as answer]
(Moving this over to the layercake repo and converting it to a discussion...)

This is more of a question on how to do animations in Svelte since there's nothing LayerCake specific. Here's an example: https://svelte.dev/repl/846da111e43d45e6809433c59c5a37a8?version=3.46.2

We create a Svelte tween and set our new data to that. Svelte handles interpolating between the data values according to the directions you give it.

Since the scales in LayerCake are all created based on the data, all you have to do is pass in a new value of the tweened object and everything will be calculated downstream from that.

**Comment by x4080** [comment]
Thanks for your answer
